src/prefect/_internal/pydantic/v2_schema.py

```diff
@@ -30,16 +30,26 @@ def is_v2_model(v) -> bool:
     return False
 
 
-def has_v2_model_as_param(signature: inspect.Signature) -> bool:
+def is_v2_type(v) -> bool:
+    if is_v2_model(v):
+        return True
+
+    try:
+        return v.__module__.startswith("pydantic")
```

Comment:
I think this would be true for any object imported from `pydantic`:

```python
>>> from pydantic.v1 import SecretStr
>>> SecretStr.__module__
'pydantic.v1.types'
```

and 
```python
>>> from pydantic import SecretStr
>>> SecretStr.__module__
'pydantic.types'
```

Maybe it'll be more reliable to check for equality? i.e.
`if HAS_PYDANTIC_V2 and v.__module__ == 'pydantic.types'`?

Reply:
good catch! updated in [116001c](https://github.com/PrefectHQ/prefect/pull/11656/commits/116001ce16b90c4b480b2f6f5059d61fa8232ece)